### ISSUE TITLE
Makefile for setting up project

### ISSUE THREAD

**Issue by radhay-samagra** [Samagra-Development/PDF-Package#34]
# Description
Add a Makefile for setting up the project and containers along with stages for prefilling dummy data.

**Comment by RyanWalker277** [comment]
Hey @radhay-samagra @ChakshuGautam ! Just wanted to know about the steps to prefill dummy-data in the database.Apart from this step , I am done with the makefile.Do let me know , I will make a PR soon as soon as I complete the prefilling part at my end and add it to the makefile.

**Comment by radhay-1199** [comment]
@RyanWalker277, I don't see any delta in creating dummy data in DB. dummy-data is just the config that needs to be there in Django admin for selecting shortener and templater service. We can have a step in the makefile for creating these entries in `Generic Config` table with the value from microservices that will be deployed from `docker-compose.gitpod.yml` file.
ex:
=> entry in generic config table for:
```json
{
  "MINIO_HOST": "YOUR MINIO HOST WITH PORT", // host:port
  "MINIO_ACCESS_KEY": "", // minio/s3 username for login
  "MINIO_SECRET_KEY": "", // minio/s3 password for login
  "MINIO_BUCKET_NAME": "doc-generator", // name of the minio bucket used for storing docs
  "SHORTENER_URL": "{{yaus_host_address}}/sr/addURL" // can be any url shortener service url
}
```
cc @ChakshuGautam

**Comment by RyanWalker277** [comment]
> @RyanWalker277, I don't see any delta in creating dummy data in DB. dummy-data is just the config that needs to be there in Django admin for selecting shortener and templater service. We can have a step in the makefile for creating these entries in `Generic Config` table with the value from microservices that will be deployed from `docker-compose.gitpod.yml` file. ex: => entry in generic config table for:
> 
> ```json
> {
>   "MINIO_HOST": "YOUR MINIO HOST WITH PORT", // host:port
>   "MINIO_ACCESS_KEY": "", // minio/s3 username for login
>   "MINIO_SECRET_KEY": "", // minio/s3 password for login
>   "MINIO_BUCKET_NAME": "doc-generator", // name of the minio bucket used for storing docs
>   "SHORTENER_URL": "{{yaus_host_address}}/sr/addURL" // can be any url shortener service url
> }
> ```
> 
> cc @ChakshuGautam

The process is more clear to me now. But on setting up the minio instance locally and filling all the values in the `Generic Config` table , still I am unable to make the api calls.It throws the error `Something went wrong: failed to resolve broker hostname` on making the api call.Did I miss something?

**Comment by radhay-1199** [comment]
I think there is issue in your config provided for rabbitmq

**Comment by FirePing32** [comment]
Hey @radhay-1199 
I am setting up the project locally. Should I follow the instructions in the readme, or setup using Docker?

**Comment by radhay-1199** [comment]
@yuvrajsab please guide @FirePing32  here

**Comment by yuvrajsab** [comment]
hey @FirePing32, I'm assuming you are talking about the current nest.js app that is in development right now. So for setting up the project locally, we recommend you use docker, it is a single command deployment. All you have to do is create a .env file by using this command `cp .env.copy .env` and start the container using `docker-compose -f docker-compose.local.yml up -d` and it will start the project. You can visit `localhost:3000` to confirm this.
We need to update the readme to add installation instructions till then just follow this.

**Comment by FirePing32** [comment]
Hey @yuvrajsab 
I have set up the project locally. 
Should the steps which you described above be included in the Makefile ?

**Comment by yuvrajsab** [comment]
@FirePing32 i don't think we need makefile anymore. because our setup is totally dependent on docker now and we can write instructions of whatever we need on readme itself. Cc: @radhay-1199

**Comment by FirePing32** [comment]
Hmm
So should I update the readme to include the docker commands ?

**Comment by yuvrajsab** [comment]
@FirePing32 yes please, create a github issue for this and then raise a pr. add installation instructions to the readme.
